python/django/cockroach_example/views.py

```diff
@@ -2,16 +2,34 @@
 from django.utils.decorators import method_decorator
 from django.views.generic import View
 from django.views.decorators.csrf import csrf_exempt
+from django.db import IntegrityError
 
 import json
 import sys
+import time
 
 from .models import *
 
 class PingView(View):
     def get(self, request, *args, **kwargs):
         return HttpResponse("python/django", status=200)
 
+def retry_on_exception(num_retries=3, on_failure=HttpResponse(status=500), delay_=0.5, backoff_=1.5):
+    def retry(view):
+        def wrapper(*args, **kwargs):
+            delay = delay_
+            for i in range(num_retries):
+                try:
```

Comment:
for an error to get raised, you're going to need to wrap the execution of `view` in a `with transaction.atomic():` (`from django.db import transaction`).

Are we trying to support nested retries here? if so, this won't work right?